src/cors.ts

```diff
@@ -0,0 +1,48 @@
+import type {CorsOptions, CorsRequest} from "cors";
+import * as cors from "cors";
+import fetch from "node-fetch";
+
+
+const LOCALHOST_ORIGIN = "https://localhost:8080";
+const ROOKOUT_ORIGIN_REGEX = /^https:\/\/.*\.rookout(?:-dev)?\.com$/;
+const DYNATRACE_ORIGIN_REGEX = /^https:\/\/.*\.dynatrace(?:labs)?\.com$/;
+
+const ALLOW_CORS_OPTION: CorsOptions = {origin: true};
+const DENY_CORS_OPTION: CorsOptions = {origin: false};
+
+
+const verifiedOriginsCache = new Set([LOCALHOST_ORIGIN]);
+
+
+const corsOptionsDelegate = async (req: CorsRequest, callback: (err: Error | null, options?: CorsOptions) => void) => {
+    try {
+        const origin = req.headers.origin;
+        if (verifiedOriginsCache.has(origin) || ROOKOUT_ORIGIN_REGEX.test(origin)) {
+            callback(null, ALLOW_CORS_OPTION);
+            return;
+        }
+
+        if (!DYNATRACE_ORIGIN_REGEX.test(origin)) {
+            callback(null, DENY_CORS_OPTION);
+            return;
+        }
+
+        const response = await fetch(`${origin}/platform-reserved/dob/isapprefallowed?appOrigin=${origin}`);
```

Comment:
Let's add an env variable to skip the HTTP check
@ElDuderinos

Reply:
Also, let's cache the result for a while :)

Reply:
The results are cached (if successful), take a look at line 37.
I didn't want to cache failed results because it might be because of a network issue or timeout or something so I didn't want to prevent it from retrying.

Reply:
@LiranLast @ElDuderinos As a packaged Electron app can't read system env variables during runtime, I added an option in the debug menu (when you right click the X) to toggle skipping verification. Should I change the text to something more clear?

Reply:
@ron-rookout It may be possible to read env using a package, I remember running into it while patching lang servers

Believe it is called electron-env

Also note this problem mainly occurs in macOs if I recall correctly

Reply:
@Urook I didn't find this one specifically, I have searched for other ways and couldn't find any either